server/server_test.go

```diff
@@ -341,3 +341,36 @@ func runTestMultiPacket(c *C) {
 		}
 	})
 }
+
+func runTestMultiStatements(c *C) {
+	runTests(c, dsn, func(dbt *DBTest) {
+		// Create Table
+		dbt.mustExec("CREATE TABLE `test` (`id` int(11) NOT NULL, `value` int(11) NOT NULL) ")
+
+		// Create Data
+		res := dbt.mustExec("INSERT INTO test VALUES (1, 1)")
+		count, err := res.RowsAffected()
+		c.Assert(err, IsNil, Commentf("res.RowsAffected() returned error"))
+		c.Assert(count, Equals, int64(1))
+
+		// Update
+		res = dbt.mustExec("UPDATE test SET value = 3 WHERE id = 1; UPDATE test SET value = 4 WHERE id = 1; UPDATE test SET value = 5 WHERE id = 1;")
```

Comment:
That's difficult.  

Actually, the mysql library in Go is not designed for multiple statements, mysql.Query result a
`sql.Result`, rather than `[]sql.Result`.
For example, how to represent the result of this query?

```
"SELECT value FROM test WHERE id=1; SELECT value, id FROM test;"
```

--------------------------

So, if you want dig the Rabbit Hole, that's see something more interesting...

mysql protocol has a concept of "capability", which is used to make an agreement with client in the handshake packet.

The library design don't support multiple result  at all, so the Go mysql library won't set MULTI_RESULTS by default.

To change that behavior, you should set 	

```
dsn += "&multiStatements=true"
```

But you don't see it in the test code, why? 
It involve a bug of TiDB, which would be fixed in this PR https://github.com/pingcap/tidb/pull/1639